File: lab4/src/container/ListFlatContainer.java
package container;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;
import data.Flat;

public class ListFlatContainer implements FlatsContainer {

    private final List<Flat> flats;

    public ListFlatContainer(List<Flat> flats) {
        this.flats = new ArrayList<>(flats);
    }

    @Override
    public List<Flat> findByArea(double lowerArea, double upperArea) {
        return this.flats.stream()
                .filter(s -> lowerArea >= s.getArea() && upperArea <= s.getArea())
                .collect(Collectors.toList());
    }

    @Override
    public void add(Flat flat) {
        this.flats.add(flat);
    }

    @Override
    public String toString() {
        return this.flats.stream()
                .map(Object::toString)
                .collect(Collectors.joining("\n"));
    }
}

File: lab6/src/container/ListFlatContainer.kt
package container

import data.Flat

class ListFlatContainer(flats: List<Flat>) : FlatsContainer {

    private val flatsList: MutableList<Flat> = ArrayList(flats)

    override fun findByArea(lowerArea: Double, upperArea: Double): List<Flat> {
        return flatsList.filter { flat ->
            flat.area >= lowerArea && flat.area <= upperArea
        }
    }

    override fun add(flat: Flat) {
        flatsList.add(flat)
    }

    override fun toString(): String {
        return flatsList.joinToString("\n") { it.toString() }
    }
}
